coremltools/converters/mil/frontend/tensorflow/test/test_ops.py

```diff
@@ -2198,38 +2198,38 @@ class TestEinsum(TensorFlowBaseTest):
             compute_units,
             backends,
             [
+                # Hardcoded cases
                 "abcd,adce->abce",
                 "abc,cbd->abd",
                 "bnqd,bnkd->bnqk",
                 "abc,cd->abd",
                 "abc,cde->abde",
                 "btnh,bfnh->bnft",
                 "bnft,btnh->bfnh",
-                "abcd,cde->abe"
+                "abcd,cde->abe",
+                # Generic cases
+                "i,i->i",
+                "i,j->ij",
+                "ab,b->a",
+                "ab,ab->b",
+                "abc,abc->a",
+                "abc,abc->c",
+                "abc,bac->c",
+                "abc,acd->abd",
+                "abc,abc->ab",
+                "abc,abc->bc",
+                "abc,bac->ba",
+                "abcd,cb->dca",
+                "abcd,acdb->ad",
+                "abcd,abde->abce",
+                "abcd,efbd->eafc",
+                "acdb,bade->abce",
             ],
             [False, True],
         )
     )
-    def test(self, compute_unit, backend, equation, reverse_input_order):
-        
-        if equation == "abcd,adce->abce":
-            input_shapes = [[3, 4, 2, 6], [3, 6, 2, 2]]
-        elif equation == "abc,cbd->abd":
-            input_shapes = [[4, 2, 6], [6, 2, 2]]
-        elif equation == "bnqd,bnkd->bnqk":
-            input_shapes = [[1, 2, 3, 4], [1, 2, 4, 4]]
-        elif equation == "abc,cd->abd":
-            input_shapes = [[2, 3, 4], [4, 5]]
-        elif equation == "abc,cde->abde":
-            input_shapes = [[2, 3, 4], [4, 5, 6]]
-        elif equation == "btnh,bfnh->bnft":
-            input_shapes = [[1, 2, 3, 4], [1, 5, 3, 4]]
-        elif equation == "bnft,btnh->bfnh":
-            input_shapes = [[1, 2, 3, 4], [1, 4, 2, 6]]
-        elif equation == "abcd,cde->abe":
-            input_shapes = [[1, 2, 3, 4], [3, 4, 6]]
-        else:
-            raise ValueError("unrecognized equation")
+    def test(self, use_cpu_for_conversion, backend, equation, reverse_input_order):
```

Comment:
please leave the `compute_unit`

Reply:
Ah, sorry.
I forgot to push last commit.